src/witness/yamlWitness.ml

```diff
@@ -193,15 +193,25 @@ struct
     in
     let task = Entry.task ~input_files ~data_model ~specification in
 
+    let is_stub_node n =
+      let fundec = Node.find_fundec n in
+      Cil.hasAttribute "goblint_stub" fundec.svar.vattr
+    in
+
     let is_invariant_node (n : Node.t) =
       let loc = Node.location n in
       match n with
-      | Statement _ when not loc.synthetic && WitnessInvariant.is_invariant_node n -> true
+      | Statement _ when not loc.synthetic && WitnessInvariant.is_invariant_node n ->
+        not (is_stub_node n)
```

Comment:
Any particular reason why this is not added to the `when` clause above?

Reply:
I probably did it for line length reasons, but I now refactored it to just have all the conditions in the case body.